integrations/aws/main.py

```diff
@@ -33,21 +37,49 @@
 )
 
 
+async def _handle_global_resource_resync(
+    kind: str,
+    credentials: AwsCredentials,
+    current_session: aioboto3.Session,
+    handle_exceptions: bool = True,
+) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    try:
+        async for batch in resync_cloudcontrol(kind, current_session):
+            yield batch
+    except Exception as e:
+        if is_access_denied_exception(e):
+            if handle_exceptions:
+                logger.info(f"Trying to resync {kind} in all regions until success")
+                async for session in credentials.create_session_for_each_region():
+                    s = await session
+                    try:
+                        async for batch in resync_cloudcontrol(kind, s):
+                            yield batch
+                        break
+                    except Exception as e:
```

Comment:
I am less of a fan of doing such a complicated logics inside error handling, my suggestion would be to return an value to indicate that it failed on authorization and if that was received then act on performing the logic inside of the current `except` block